### PR TITLE
Set color

### LINKED ISSUE #5: Select a color from a dropdown menu of colors

### LINKED ISSUE #6: Click on a single cell, changing its color to the currently selected color

### PR DESCRIPTION
**Pull request by wongjessica**
closes #5, #6
- implemented set color function
- added dropdown color menu to match functions
- added comments

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -107,25 +110,37 @@ const applyToCells = callback => {
...
 
 // CLEAR ALL -----------------------------
 const clearAll = () => {
   // clear color regardless of prev color
   applyToCells(col => (col.style.backgroundColor = ""));
 };
+
+//Function to color the cell on user prompt
+function colorChange(){
+  console.log(this);
```
Comment (khinshankhan) [inline script.js]: remove debugging log statement

File: script.js [hunk 2]
```diff
@@ -107,25 +110,37 @@ const applyToCells = callback => {
   }
 };
 
+//Function to implement the Color Selector
+let colorSelector = document.getElementById("color_choose"), color_change;
```
Comment (khinshankhan) [inline script.js]: This is a bit confusing. It's the equivalent of 
```
let colorSelector, color_change;
colorSelector = document.getElementById("color_choose");
```
But the one liner is much harder to read. Usually it's preferred to use `const` for non changing values.

File: script.js [hunk 3]
```diff
@@ -32,6 +33,7 @@ const addCol = () => {
   // edge case: there are no rows to add a cell to
   if (rows == 0) {
     const row = document.createElement("tr");
+    cell.addEventListener("click",colorChange);
```
Comment (khinshankhan) [inline script.js]: This line makes no sense, there is no cell in this scope.

File: script.js [hunk 4]
```diff
@@ -107,25 +110,37 @@ const applyToCells = callback => {
...
 };
 
 // CLEAR ALL -----------------------------
 const clearAll = () => {
   // clear color regardless of prev color
   applyToCells(col => (col.style.backgroundColor = ""));
 };
+
+//Function to color the cell on user prompt
+function colorChange(){
```
Comment (khinshankhan) [inline script.js:141]: Function hoisting works, however it's a bit confusing. It should be preferred to move the logic and logic that it depends on to the top of the file, before the logic that depends on it. You can keep this if you'd like though.

File: script.js (hunk 2 above)
Comment (khinshankhan) [reply script.js]: What is the default value of `color_change`? `colorSelect` is fired on an onchange event. You should either manually set the value to the default color in the selector, or better yet use the value of the element you get, `colorSelector`.

File: script.js (hunk 1 above)
Comment (khinshankhan) [reply script.js]: also remove lines 3-4 if you're going to use `color_change`

**Comment by wongjessica** [comment]
Fixed recommendated changes, please review @kkhan01

**Review by khinshankhan — APPROVED** [review]

**Pull request merged**